library/core/src/slice/mod.rs

```diff
@@ -90,11 +101,21 @@ impl<T> [T] {
     #[rustc_const_stable(feature = "const_slice_len", since = "1.32.0")]
     #[inline]
     // SAFETY: const sound because we transmute out the length field as a usize (which it must be)
-    #[allow_internal_unstable(const_fn_union)]
+    #[allow_internal_unstable(const_fn_union, const_unreachable_unchecked)]
     pub const fn len(&self) -> usize {
         // SAFETY: this is safe because `&[T]` and `FatPtr<T>` have the same layout.
         // Only `std` can make this guarantee.
-        unsafe { crate::ptr::Repr { rust: self }.raw.len }
+        let raw_len = unsafe { crate::ptr::Repr { rust: self }.raw.len };
+
+        match mem::size_of::<T>() {
+            0 => {}
+            // SAFETY: this branch taken if mathematically `isize::MAX < raw_len * N`.
+            // But references must point to one allocation with size at most isize::MAX.
+            n if (isize::MAX as usize) / n < raw_len => unsafe { unreachable_unchecked!() },
```

Comment:
I mean, you could use the function directly now without the helper macro.

Reply:
I wasn't aware this already got fixed :slightly_smiling_face:

Reply:
It haven't fixed, just that one function only support one `allow_internal_unstable` annotation.
That's just my guess.

Reply:
Okay, fixed.